lib/queryBuilder/operations/eager/RelationJoinBuilder.js

```diff
@@ -575,16 +575,16 @@ function createModifierQuery({ builder, modelClass, expr, modifiers, relation })
 
   for (let i = 0, l = expr.$modify.length; i < l; ++i) {
     const modifierName = expr.$modify[i];
-    let modifier = null;
+    const modifier = createModifier({
+      modifier: modifierName,
+      modelClass: relation.relatedModelClass,
+      modifiers
+    });
 
     try {
-      modifier = createModifier({
-        modifier: modifierName,
-        modelClass: relation.relatedModelClass,
-        modifiers
-      });
+      modifier(modifierQuery);
     } catch (err) {
-      if (err instanceof ModifierNotFoundError) {
+      if (err instanceof modelClass.ModifierNotFoundError) {
```

Comment:
@koskimas this is why the tests were failing: The exceptions are now thrown when using the modifier, not when creating it.